cmd/clusterctl/client/cluster/template_test.go

```diff
@@ -225,6 +240,51 @@ func Test_templateClient_getGitHubFileContent(t *testing.T) {
 	}
 }
 
+func Test_templateClient_getRawUrlFileContent(t *testing.T) {
+	type args struct {
+		rURL string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "Return custom template",
+			args: args{
+				rURL: "https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/pods/commands.yaml",
+			},
+			want:    []byte(rawURLTemplate),
+			wantErr: false,
+		},
+		{
+			name: "Wrong url",
+			args: args{
+				rURL: "https://foo.com/path/to/cluster-template.yaml",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			c := &templateClient{}
+			got, err := c.getRawUrlFileContent(tt.args.rURL)
```

Comment:
Could we use a fake http client for this test?

I would really prefer if our unit tests are not doing Get requests against GitHub (and they don't depend on a file there existing)

Reply:
Thanks for pointing :)